src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the.NET Foundation under one or more agreements.
+// The.NET Foundation licenses this file to you under the MIT license.
+// See the License.txt file in the project root for more information.
+
+namespace Microsoft.RoslynTools.Commands;
+
+using System.CommandLine.Invocation;
+using System.CommandLine;
+using Microsoft.RoslynTools.Authentication;
+
+internal static class PRTaggerCommand
+{
+    private static readonly PRTaggerCommandDefaultHandler s_prTaggerCommandHandler = new();
+
+    private static readonly Option<string> ProductName = new(new[] { "--product-name", "-p" }, "Name of product (e.g. 'roslyn' or 'razor')") { IsRequired = true };
+    private static readonly Option<string> VSBuild = new(new[] { "--build", "-b" }, "VS build number") { IsRequired = true };
+    private static readonly Option<string> CommitId = new(new[] { "--commit", "-c" }, "VS build commit") { IsRequired = true };
+    private static readonly Option<string> GitHubToken = new(new[] { "--github-token" }, "Token used to authenticate GitHub.") { IsRequired = true };
+    private static readonly Option<string> DevDivAzDOToken = new(new[] { "--devdiv-azdo-token" }, "Token used to authenticate to DevDiv Azure DevOps.") { IsRequired = true };
+    private static readonly Option<string> DncEngAzDOToken = new(new[] { "--dnceng-azdo-token" }, "Token used to authenticate to DncEng Azure DevOps.") { IsRequired = true };
```

Comment:
I think we should handle these tokens the same way `VSBranchInfo` does - https://github.com/dotnet/roslyn-tools/blob/main/src/dotnet-roslyn-tools/Commands/VSBranchInfoCommand.cs#L43. Add the token options from CommonOptions and then parse the setting from the `ParseResult`. Invalid/missing tokens could be checked up front or by simply allow the tagging to throw an exception with authorization issues occur.